pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1449,8 +1450,24 @@ func convertGateways(r ConfigContext) ([]config.Config, map[parentKey]map[k8s.Se
 			if len(addressesToReport) == 0 {
 				// There are no external addresses, so report the internal ones
 				// TODO: should we always report both?
-				addressesToReport = internal
 				addrType = k8s.HostnameAddressType
+				for _, hostport := range internal {
+					svchost, _, _ := net.SplitHostPort(hostport)
+					svc := r.Context.ps.ServiceIndex.HostnameAndNamespace[host.Name(svchost)][obj.Namespace]
+					if svc.Attributes.Type != corev1.ServiceTypeLoadBalancer {
+						// Add internal hostname if not LoadBalancer type service and not duplicate
+						found := false
+						for _, addr := range addressesToReport {
```

Comment:
seems like we should have a warning like "Assigned to service(s) %s, but address is pending: %s" around L1420